cpp/ycm/tests/CMakeLists.txt

```diff
@@ -77,13 +82,39 @@ add_executable( ${PROJECT_NAME}
                 ${SOURCES}
               )
 
+if ( MSVC )
+  # This is needed to compile tests with the gtest shared library
+  set_target_properties( ${PROJECT_NAME}
+                         PROPERTIES
+                         COMPILE_DEFINITIONS
+                         "GTEST_LINKED_AS_SHARED_LIBRARY=1" )
+
+  # Preprocessor definition used to export symbols for gmock tests
+  set_target_properties( ${SERVER_LIB} PROPERTIES DEFINE_SYMBOL "TESTS_EXPORTS" )
+
+  # Fix gtest build on MSVC 11. See:
+  # http://stackoverflow.com/questions/8274588/c2977-stdtuple-too-many-template-arguments-msvc11
+  if ( MSVC_VERSION EQUAL 1700 )
+    add_definitions( /D _VARIADIC_MAX=10 )
+  endif()
+
+  # Build gmock and ycm_core_tests targets in cmake ycm/tests folder
+  foreach( OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES} )
+    string( TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG )
+    set_target_properties( ${GMOCK_LIBRARIES} PROPERTIES
+      RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${PROJECT_BINARY_DIR} )
+    set_target_properties( ${PROJECT_NAME} PROPERTIES
+      RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${PROJECT_BINARY_DIR} )
+  endforeach()
+endif()
+
 target_link_libraries( ${PROJECT_NAME}
+                       ${Boost_LIBRARIES}
```

Comment:
That looks like BoostParts is getting doubly-linked, which is bad. If adding that line fixes it, just go with that.